### ISSUE TITLE
resolve data inconsistencies

### ISSUE THREAD

**Issue by dougkdev** [FreeUKGen/FreeCENMigration#154]
Some PARMS.dat files and .VLD files have inconsistencies in the data that prevent them from loading properly in FC2. The inconsistencies are also present in FC1, but due to less error-checking on FC1, they get loaded into the database with the errors. We need to let Brenda know which ones so they can be fixed.

**Comment by dougkdev** [comment]
(we need to also let her know what the errors are, not just which ones have errors)

**Comment by dougkdev** [comment]
I am moving this into the current sprint because it is holding up Brenda's testing (she needs NTT to test and some of the data errors are in NTT).

**Comment by dougkdev** [comment]
I have found several inconsistencies in the freecen1 data (which we import into FC2). The inconsistencies are listed below.  Some of them are data errors with the files themselves (probably to be fixed by @FreecenBren or her coordinators), and some appear to be due to incorrect placement of files in the directory structure on the production server.  I am still a little fuzzy on the exact process of how the freecen1 files are put into specific directories on the server: whether they are checked into RCS and then checked out from RCS onto the servers, or whether they are simply copied into the locations on the filesystem.  For files that are in the wrong place, etc. should I just move them myself or does @lemon-ukgen or someone with FC1 do that?  (I'm hesitant to move FC1 files around without asking since I have not been involved with FC1 at all).

Here are the inconsistencies I found that should be resolved:

PARMS.DAT data errors: (these prevent the PARMS.DAT from loading in FC2, and also prevent any .VLD files for these year/county combinations from loading):

1) fixed/1851/DEVPARMS.DAT - blank place (district) names in pieces 1883 and 1885. In both cases, the line with a blank is also an 'a' row that shouldn't be there.
2) fixed/1861/DBYPARMS.DAT - blank place (district) name in piece 2512
3) fixed/1861/WILPARMS.DAT - blank place (district) name in piece 1309. Also the first 1309 row is a 'b' row before the 'a' row for 1309.
4) fixed/1871/LNDPARMS.DAT - blank place name in 512 (it is also an 'a' row, shouldn't be there).
5) fixed/1871/NTTPARMS.DAT - blank place names in 3493 (it is also marked as an 'n' row instead of 'a' or 'b'), 3494 (it is also marked as an 'a' row, it shouldn't be there). 3532 (it is marked as an 'a' row, shouldn't be there), 3542 (both 'a' and 'b' rows), 3543 (both 'a' and 'b' rows), 3547 (an unexpected 'a' row)
6) fixed/1891/GLAPARMS.DAT - the file is malformed beginning at 4940
7) fixed/1891/NTTPARMS.DAT - blank place names in 2659 ('b' row), 2660 ('a' row), also the file is malformed beginning at 2719


PARMS.DAT file inconsistencies:

1) there are two identical files (fixed/1861/DURPARMS.DAT and fixed/1861/durparms.dat). This doesn't cause an error because we ignore the lowercased filename file, but shows up as a warning in the processing log.
2) several files that are named ctyPARMT.DAT instead of ctyPARMS.DAT (T instead of S before the dot).  I'm guessing these are templates for the county parms files given to the coords before they add all the piece and place info? We ignore these, so not a major concern, just causes warning messages in the log. The ones there as of today are:
  1871/WRYPARMT.DAT
  1891/CAEPARMT.DAT
  1891/CGNPARMT.DAT
  1891/HEFPARMT.DAT
  1891/HUNPARMT.DAT
  1891/RADPARMT.DAT
 

.VLD filename/piece inconsistencies:

1) pieces/ABD/RS755168.VLD -- piece/par_num 168/55 not in parms. This appears to actually be ABD data, but this filename doesn't correspond to any piece/par_num specified in the parms.dat
2) pieces/ABD/RS756168.VLD -- piece/par_num 168/56 not in parms
3) pieces/ARL/HS512539.VLD -- piece/par_num 539/12 not in parms
4) pieces/HAM/RG094485.VLD -- piece/par_num not in parms
5) pieces/SOM/rg094497.vld -- ? doesn't seem to be an SOM piece, not sure where it should be instead of SOM


.VLD file location inconsistencies:

1) pieces/ABD/HS410899.VLD (piece not found in parms) -- I believe this is actually a WIG file and should be in the WIG directory?
2) pieces/ABD/HS510894.VLD (piece not found in parms) -- I believe this is actually a WIG file and should be in the WIG directory?
3) pieces/ABD/HS511319.VLD -- I believe should be in ANS, not ABD?
4) pieces/ABD/RS711137.VLD -- MOR? instead of ABD
5) pieces/BDF/RG121243.VLD -- HUN? instead of BDF
6) pieces/DUR/RG104248.VLD -- WRY? instead of DUR
7) pieces/ERY/RG093220.VLD -- WRY? instead of ERY
8) pieces/ERY/rg121413.vld -- ESS? instead of ERY
9) pieces/FLN/rg122870.vld -- piece/par_num not in parms
10) pieces/FLN/RG124611.VLD -- DEN? instead of FLN
12) pieces/FLN/RG124626.VLD -- DEN? instead of FLN
13) pieces/FLN/RG124627.VLD -- DEN? instead of FLN
14) pieces/LAN/RG103233.VLD -- LEI? instead of LAN
15) pieces/LAN/RG103293.VLD -- LEI? instead of LAN
16) pieces/LAN/RG103295.VLD -- LEI? instead of LAN
17) pieces/LAN/RG122511.VLD -- LEI? instead of LAN
18) pieces/OKI/hs410002.vld -- SHI instead of OKI
19) pieces/OKI/hs410005.vld -- SHI instead of OKI
20) pieces/OKI/hs410006.vld -- SHI instead of OKI
21) pieces/OKI/hs410008.vld -- SHI instead of OKI
22) pieces/OKI/hs410009.vld -- SHI instead of OKI
23) pieces/OKI/hs410010.vld -- SHI instead of OKI
24) pieces/OKI/hs410011.vld -- SHI instead of OKI
25) pieces/OKI/hs411001.vld -- SHI instead of OKI
26) pieces/OKI/hs411003.vld -- SHI instead of OKI
27) pieces/OKI/hs411004.vld -- SHI instead of OKI
28) pieces/OKI/hs411007.vld -- SHI instead of OKI
29) pieces/OKI/hs411012.vld -- SHI instead of OKI
30) pieces/OKI/hs412001.vld -- SHI instead of OKI
31) pieces/OKI/hs412003.vld -- SHI instead of OKI
32) pieces/OKI/hs412004.vld -- SHI instead of OKI
33) pieces/OKI/hs412007.vld -- SHI instead of OKI
34) pieces/OKI/hs412012.vld -- SHI instead of OKI
35) pieces/OKI/hs413007.vld -- SHI instead of OKI
36) pieces/OKI/hs413012.vld -- SHI instead of OKI
37) pieces/OKI/hs414012.vld -- SHI instead of OKI
38) pieces/SFK/RG121561.VLD -- NFK? instead of SFK
39) pieces/SSX/RG121437.VLD -- SFK? instead of SSX
40) pieces/STS/RG090767.VLD -- MDX? instead of STS
41) pieces/STS/RG092063.VLD -- WOR? instead of STS
42) pieces/STS/RG092064.VLD -- WOR? instead of STS
43) pieces/WRY/RG093588.VLD -- ERY? instead of WRY
44) pieces/WRY/RG093594.VLD -- ERY? instead of WRY
45) pieces/WRY/RG093595.VLD -- ERY? instead of WRY
46) pieces/WRY/RG093596.VLD -- ERY? instead of WRY
47) pieces/WRY/RG104788.VLD -- ERY? instead of WRY
48) pieces/WRY/RG104790.VLD -- ERY? instead of WRY
49) pieces/WRY/RG104798.VLD -- ERY? instead of WRY
50) pieces/WRY/RG104799.VLD -- ERY? instead of WRY
51) pieces/WRY/RG104800.VLD -- ERY? instead of WRY
52) pieces/NOR/RG121524.VLD -- NOR is not a valid chapman code, should be in pieces/NFK instead of pieces/NOR/
53) directory pieces/ANG -- ANG is not a valid chapman code
54) directory pieces/LDN -- LDN is not a valid chapman code
55) directory pieces/NOR -- NOR is not a valid chapman code
56) file pieces/PEE/PEEPARMS.DAT should not be in this directory.  NOTE: it is not identical to any of the fixed/18?1/PEEPARMS.DAT files, so we need to figure out which is the correct version of the file and make sure the correct version is in the proper fixed/18?1/ directory before deleting/moving this or replacing any in the fixed directory with this one.
57) unexpected .CSV file: pieces/RFW/HS414564.csv (there is also a .VLD for this piece in the directory)
58) unexpected .CSV file: pieces/RFW/HS516559.csv (there is also a .VLD)
59) unexpected .CSV file: pieces/RFW/HS518564.CSV (there is also a .VLD)
60) unexpected .CSV file: pieces/SOM/H0511922.csv (there is also a .VLD)

**Comment by FreecenBren** [comment]
The PARMS files for the 1891 Census were all done at the start of FreeCEN and not by the COORDS who are here today. This is the Census year that FreeCEN frost started.

In fact many COORDS did not know how to do them and they were done for them.
I think the late Bob Muchamore and his team looked after that part.



When FreeCEN first started PIECES were often cross-over with neighbouring Counties.

The NTT/DBY 1891 is just one of these.
RG122719 was with the NTT Micro Fiche and was therefore was completed by that County.

When the DBY fiche came along that started with RG122720 as per the Fiche they were sent.

Many of them for that Census year were completed a few years back including the NTT 1891.
These match the PARMS for each County?

 

When the Census images were sent for the 1891 they were all on Micro Fiche.

The LDS would send a ‘County’ and the PARMS were done often from the Fiche.

The luxury we have to day to go onto the National archives Web Site example here http://discovery.nationalarchives.gov.uk/results/r?_q=RG+12%2F2659 was none existent.
So do we now change the PARMS to match the National Archives even though NTT 1891 is completed.

 

OK That is a little history.

What we need is what we are waiting for. The ability to be able to amend PARMS files online.

That is what Doug is working on. Not sure how far we have got with that?
Alternatively should I do it the old way and amend them manually and get them replaced online?

 

OK. I have picked a few others queries to do for now.

1) pieces/ABD/RS755168.VLD -- piece/par_num 168/55 not in parms. 
This appears to actually be ABD data, but this filename doesn't correspond to any piece/par_num specified in the parms.dat  
 This one I do know the answer. 
It needs being removed as it was named incorrectly as RS755168 and it has been replaced by RS756168 which had been wrongly names. See number 2 below.
2) pieces/ABD/RS756168.VLD -- piece/par_num 168/56 not in parms

3) pieces/ARL/HS512539.VLD -- piece/par_num 539/12 not in parms
This one is for Scotland.  It could well be named incorrectly as 539 is in the PARMS.
I will have to check with Rena on this.


SCT1851/539

Jura & Colonsay

Jura & Colonsay


4) pieces/HAM/RG094485.VLD -- piece/par_num not in parms
This needs removing from HAM as it is part of EWS and is online under EWS.
Why this is there I am not sure as it is not part of the HAM Parms for 1861.

5) pieces/SOM/rg094497.vld -- ? doesn't seem to be an SOM piece, not sure where it should be instead of SOM

This needs removing from SOM as it is part of EWS and is online under EWS.
Why this is there I am not sure as it is not part of the SOM Parms for 1861.

 

The DEN/FLN ones are of no surprise.
They have towns split between them and are forever changing counties.
These came on CD and are mixed together.

57) unexpected .CSV file: pieces/RFW/HS414564.csv (there is also a .VLD for this piece in the directory)
58) unexpected .CSV file: pieces/RFW/HS516559.csv (there is also a .VLD)
59) unexpected .CSV file: pieces/RFW/HS518564.CSV (there is also a .VLD)
60) unexpected .CSV file: pieces/SOM/H0511922.csv (there is also a .VLD)

These above need removing.   When I send Lemon VLD’s they are sent in 2 folders.

1.   One for new VLD’s per county to be uploaded.

2.   The other one for Amended VLD’s
This folder means that there is already a VLD been uploaded previously.
This need removing first and then the Amended one needs to be added.
This looks like there might be some historic ones lurking about before Lemon took on the job of doing FreeCEN uploads. It looks like some CSV’s were sent and also manged to get uploaded.
This needs to be addressed so that only VLD’s are uploaded and anything other than them should be rejected.

 

Sorry it is the granddaughters 18th birthday and we are now off to the party.
If you let me have a list of the ones that you need me to have a look at and do something with please send me an email and I will see what I can find.

 

Brenda

**Comment by helhyde** [comment]
Hi Brenda,

Thanks for this.  I think we should work closely with TNA on this - e.g.
routing people to their pages for the explanation, much as we do to GENUKI
from FreeREG for C of E Parishes.

We need to get the TNA to say 'high quality transcriptions may be
available, for free, from FreeCEN' on all their pages with a link to us,
ideally above the link to Ancestry.

I've started a shopping list to take to TNA....

Best wishes,

Pat

*- -*

*Dr Pat Reynolds*
Executive Director
Free UK Genealogy  <http://www.freeukgenealogy.org.uk/>
A charity registered in England and Wales, number 1096940, transitioning to
a Charitable Incorporated Organisation registered in England and Wales,
number 1167484
VAT registration: 233 0105 70

​+44 ​1904 541411​ +44 7943 145387

36 Albemarle Road, York, YO23 1ER, UK

**Comment by PatReynolds** [comment]
dependent on #99

**Comment by lemon-ukgen** [comment]
Answering some of @dougkdev's question about the FreeCEN1 files, as much a know:

### The `PARMS` files

( eg `~apache/hosts/freecen/fixed/1891/AGYPARMS.DAT`)

These live in CVS at `UKGEN-FreeCEN/fixed/...`. I have occasionally been asked to update one but very infrequently.

### The `VLD` files

( eg `/raid/freecen/pieces/ABD/HS410169.VLD` )

I've never been clear what mechanism was used to update these before I got involved but Brenda and I came with a tactical fix because updates had stalled for a while. This involves Brenda updating a Dropbox folder with new or updated `VLD` files organised in their `XYZ` subdirectories  and a script to work through these putting them into the file tree on the master server.

The script is at `~apache/hosts/freecen/bin/update-pieces.rb` and its use outlined in the `UKGEN-FreeCEN/README`) file in CVS. 

### FreeCEN1 build process

After updating the `VLD` files on the master I would run the database update process, also in the README. AIUI that combines the `PARMS` deployed in the last handoff into `~apache/hosts/freecen`) with the `VLD` in `/raid/freecen/pieces` to build the database.

So that's the process I know of to date. I reckon the `VLD` files can be fixed in place without my involvement. For consistency I think any changes to `PARMS` files should be committed back to UKGEN-FreeCEN CVS.

I am happy to help where I can but there's nothing here that I own in particular :)

**Comment by dougkdev** [comment]
Thanks, @lemon-ukgen.

**Comment by dougkdev** [comment]
I have resolved most of the inconsistencies.  I made corrections to the 7 PARMS.DAT files that had errors in them and emailed the corrected versions to @FreecenBren for review. The corrected versions are loaded on FC2 (test2), but not checked in to freecen1 CVS until approved by her. I have also relocated the approximately 60 .VLD files to the correct county directories so they load on FC2.

Three .VLD file errors remain:

1) ARL/HS512539.VLD - this piece is from Scotland and @FreecenBren  is going to check with Rena about it. (possibly named incorrectly?)

2) ABD/RS756168.VLD - This one does not load due to mismatch between file name and parms.dat information. (note: it is also listed as a file that didn't load on FC1 at http://www.freecen.org.uk/errors.html)

3) NTT/RG122683.VLD - this file has a row at the end with "C    001886000" instead of valid data.

**Comment by dougkdev** [comment]
@FreecenBren  has reviewed the 7 amended PARMS.DAT files and they have been checked in to the CVS repository to be picked up with the next freecen1 update.

**Comment by dougkdev** [comment]
It is my opinion that we can go ahead and close this story.  The large number of inconsistencies that were the reason I opened the story have been resolved (the 7 parms.dat files that wouldn't load on FC2 due to errors and the 60+ VLD files that wouldn't load on FC2 because they were in the incorrect county directories). There are a handful of questions still being looked into, but we are now at a small number of inconsistencies, about the number that might arise after the FC1 update for any given month.
